src/Type/WebMozartAssert/AssertTypeSpecifyingExtension.php

```diff
@@ -335,25 +335,33 @@ private static function getExpressionResolvers(): array
 				},
 				'isInstanceOf' => static function (Scope $scope, Arg $expr, Arg $class): ?Expr {
 					$classType = $scope->getType($class->value);
-					if (!$classType instanceof ConstantStringType) {
+					if ($classType instanceof ConstantStringType) {
```

Comment:
Isn‘t it ClassStringType instead?

Reply:
Good point, but it should be fine because the extension supports only narrowing if constant strings are used like `Assert::isInstanceOf($foo, Foo::class)`. It narrows it down to an instance of Foo then, as you can guess :)

I thought briefly about it, but adding support for ClassStringType would potentially only help to narrow the input down to `object`. I'd rather not do that here because of problems that might arise from `ImpossibleCheckTypeStaticMethodCallRule` in case an object is passed already. There recently were some issues like that and some of the open ones are similar too I think. I'll need to take a deeper look at some point.